idaes/gas_solid_contactors/unit_models/fixed_bed_0D.py

```diff
@@ -0,0 +1,355 @@
+                    doc="Solid phase energy holdup")
+
+            @self.Constraint(self.flowsheet().config.time,
+                             doc="Solid phase energy holdup constraints")
+            def solids_energy_holdup_constraints(b, t):
+                return b.solids_energy_holdup[t] == (
+                      self.volume_solid *
+                      b.solids[t].get_energy_density_terms("Sol"))
+
+            self.solids_energy_accumulation = DerivativeVar(
+                        self.solids_energy_holdup,
+                        doc="Solids energy accumulation")
+
+            @self.Constraint(self.flowsheet().config.time,
+                             doc="Solid phase energy accumulation constraints")
+            def solids_energy_accumulation_constraints(b, t):
+                # if t == self.flowsheet().config.time.first():
+                #     return Constraint.Skip
+                # else:
+                return b.solids_energy_accumulation[t] == \
+                    - sum(b.reactions[t].reaction_rate[r] *
+                          b.volume_solid *
+                          b.reactions[t].dh_rxn[r]
+                          for r in b.config.reaction_package.
+                          rate_reaction_idx)
+
+    def initialize(blk, state_args=None, outlvl=6,
+                   solver='ipopt', optarg={'tol': 1e-6}):
+        '''
+        This is a general purpose initialization routine for simple unit
+        models. This method assumes a single ControlVolume block called
+        controlVolume, and first initializes this and then attempts to solve
+        the entire unit.
+        More complex models should overload this method with their own
+        initialization routines,
+        Keyword Arguments:
+            state_args : a dict of arguments to be passed to the property
+                           package(s) to provide an initial state for
+                           initialization (see documentation of the specific
+                           property package) (default = {}).
+            outlvl : sets output level of initialization routine
+                 * 0 = Use default idaes.init logger setting
+                 * 1 = Maximum output
+                 * 2 = Include solver output
+                 * 3 = Return solver state for each step in subroutines
+                 * 4 = Return solver state for each step in routine
+                 * 5 = Final initialization status and exceptions
+                 * 6 = No output
+            optarg : solver options dictionary object (default={'tol': 1e-6})
+            solver : str indicating which solver to use during
+                     initialization (default = 'ipopt')
+        Returns:
+            None
+        '''
+        # Set solver options
+        init_log = getInitLogger(blk.name, outlvl)
+#        solve_log = getSolveLogger(blk.name, outlvl, tag="unit")
+        opt = get_solver(solver, optarg) # create solver
+        opt.options = optarg
```

Comment:
Take a look at the initialize methods in the bubbling fluidized bed and moving bed and update accordingly. 
The initialize method here should look like:
def initialize(blk, state_args={}, outlvl=idaeslog.NOTSET,
                   solver=None, optarg={}):

Reply:
A minor correction (or more recent change in IDAES): `state_args=None` and `optarg=None` should be the default.

There is an underlying Python design decision that means that passing empty dicts (`{}`) as default arguments is a bad idea - basically the empty dict only gets created when the code is first important, not each time the code is used. Thus that empty dict can often becomes non-empty the first time the code is called.

Note that this means you need to handle this later, generally by doing something like:

```
if state_args is None:
    state_args = {}
```

In the case of `optargs`, the (new) IDAES `get_solver` mehtod accepts `optargs=None` and handles it correctly for you.